android/app/src/main/java/com/cBreez/client/BreezForegroundService.kt

```diff
@@ -0,0 +1,196 @@
+    }
+
+    // SDK events listener
+    inner class SDKListener : EventListener {
+        override fun onEvent(e: BreezEvent) {
+            Logger.tag(TAG).info { "Received event $e" }
+            if (e is BreezEvent.InvoicePaid) {
+                val pd = e.details
+                handleReceivedPayment(pd.bolt11, pd.paymentHash, pd.payment?.amountMsat)
+            }
+        }
+    }
+
+    private var breezSDK: BlockingBreezServices? = null
+    private val pollerScope = CoroutineScope(Dispatchers.Main.immediate + SupervisorJob())
+    private val binder = BreezNodeBinder()
+
+    /** List of payment hashes that are being polled */
+    private val paymentHashPollerList = mutableListOf<String>()
+
+    override fun onCreate() {
+        super.onCreate()
+        Logger.tag(TAG).debug { "Creating Breez node service..." }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            registerNotificationChannels(applicationContext)
+        }
+        Logger.tag(TAG).debug { "Breez node service created." }
+    }
+
+    // =========================================================== //
+    //                      SERVICE LIFECYCLE                      //
+    // =========================================================== //
+
+    override fun onBind(intent: Intent?): IBinder {
+        Logger.tag(TAG).debug { "Binding Breez node service from intent=$intent" }
+        paymentHashPollerList.clear()
+        stopForeground(STOP_FOREGROUND_REMOVE)
+        dismissForegroundServiceNotification(applicationContext)
+        return binder
+    }
+
+    override fun onUnbind(intent: Intent?): Boolean {
+        return false
+    }
+
+    private val shutdownHandler = Handler(Looper.getMainLooper())
+    private val shutdownRunnable: Runnable = Runnable {
+        Logger.tag(TAG).debug { "Reached scheduled shutdown..." }
+        if (paymentHashPollerList.isEmpty()) {
+            stopForeground(STOP_FOREGROUND_REMOVE)
+        } else {
+            stopForeground(STOP_FOREGROUND_DETACH)
+        }
+        shutdown()
+    }
+
+    /** Shutdown the node, close connections and stop the service */
+    private fun shutdown() {
+        Logger.tag(TAG).info { "Shutting down Breez node service" }
+        stopSelf()
+    }
+
+    // =========================================================== //
+    //                    START COMMAND HANDLER                    //
+    // =========================================================== //
+
+    /** Called when an intent is called for this service. */
+    @RequiresApi(Build.VERSION_CODES.TIRAMISU)
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        super.onStartCommand(intent, flags, startId)
+        Logger.tag(TAG).debug {
+            "Start Breez node service from intent [ intent=$intent, flag=$flags, startId=$startId ]"
+        }
+        intent.getRemoteMessage()?.let {
```

Comment:
Seems like we should schedule a shutdown here otherwise there are cases where the service will not stop.

Reply:
https://github.com/breez/c-breez/pull/748#discussion_r1439530834